core/src/util/fastmap.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <algorithm>
+
+namespace Tangram {
+
+// see also:
+// http://www.boost.org/doc/libs/1_59_0/doc/html/boost/container/flat_map.html
+// https://realm.io/assets/news/binary-search/blog.cpp
+
+template<typename K, typename T>
+struct fastmap {
+    std::vector<std::pair<K, T>> map;
```

Comment:
Yes, we should try that. It probably depends on the size of the map. This is is a first start which showed clear improvement in manual perf testing. But we could also postpone fastmap until we have a proper benchmarks.